homeassistant/components/dlna_dms/config_flow.py

```diff
@@ -74,6 +74,9 @@ async def async_step_ssdp(
             LOGGER.debug("async_step_ssdp: discovery_info %s", pformat(discovery_info))
 
         await self._async_parse_discovery(discovery_info)
+        if TYPE_CHECKING:
```

Comment:
I don't think it's necessary to check for `TYPE_CHECKING`. You can just use a bare `assert` like in `_create_entry`, as the condition will always be true, even at runtime, and the `TYPE_CHECKING` check implies there's a special condition only while the type checker is running.

Reply:
I think we should hide asserts which are only there to influence the type checker behind `TYPE_CHECKING`.
For me it doesn't imply the condition is only true when the type checker is running, but that it's not something we want to spend cycles on checking when it isn't.

Reply:
To me the asserts are also documenting to fellow programmers about what we believe to be true about the program's state. In this case, it's an assumed post-condition on the `_async_parse_discovery` method.

If we really cared about the performance impact, we could run `python3` with the `-O` flag to disable all assertions. 

Aside: how are you getting 100% code coverage of this file when the unit tests shouldn't be running with `TYPE_CHECKING` enabled?

Reply:
> Aside: how are you getting 100% code coverage of this file when the unit tests shouldn't be running with TYPE_CHECKING enabled?

They are excluded from coverage automatically.

Reply:
> > Aside: how are you getting 100% code coverage of this file when the unit tests shouldn't be running with TYPE_CHECKING enabled?
> 
> They are excluded from coverage automatically.

Hmm, that means that what's tested and what's type-checked are two different sets of code. This could lead to weird edge-cases. E.g: a maintainer changes `_async_parse_discovery` to not always set `self._name` and updates the tests to match, but doesn't change these two lines. Thus the type checker will be satisfied that `self._name is not None`, the tests will pass with it set to `None`, and who knows what happens at runtime.

> Please note, the change made here, is how we generally handle this in Home Assistant. I thus approve it.

Ok. I hope I've given you some good reasons to reconsider.

Reply:
I still don't like it, and I'm sorry I missed the earlier discussions. But since this is the path chosen, I'll stop raising the issue.